MP-4512: Add angular-components doc (#146)
spryker/spryker.github.io: docs/marketplace/dev/front-end/angular-components.md

- The decorator accepts a metadata object that gives information about the component.
- A component must belong to the NgModule in order for it to be usable by another component or application.
- Components control their runtime behavior by implementing Life-Cycle hooks.

A common pattern in Angular is sharing data between a parent component and one or more child components. You can implement this pattern by using the `@Input()` and `@Output()` decorators.

## Component Inputs

Angular Component has Inputs - these are the properties/attributes that can be passed to it from the outside. In the web component form the inputs are translated to HTML attributes and read as strings so extra parsing usually is required (we ship a set of decorators for most common types parsing e.g. `@ToJson`, `ToBoolean`...).

To define input you have to use `@Input` angular decorator that marks a class field as an input property and supplies configuration metadata. The input property is bound to a DOM property in the template. During change detection, Angular automatically updates the data property with the DOM property's value.

## Component Outputs

> Angular Component has Outputs - these are the events that the component may emit at any point in time to the outside. In the web component form outputs are translated to DOM Events.

Reviewer: ```suggestion
Angular Component has Outputs - these are the events that the component may emit at any point in time to the outside. In the web component form outputs are translated to DOM Custom Events.
```

> To define input you have to use `@Output` angular decorator that marks a class field as an output property and supplies configuration metadata. The DOM property bound to the output property is automatically updated during change detection.

Reviewer: ```suggestion
To define input you have to use `@Output` angular decorator that marks a class field as an output property and supplies configuration metadata. A callback method bound to the output property is invoked every time output is emitted.
```

## Component Lifecycle

### Hooks for the Component

`constructor`
This is invoked when Angular creates a component or directive by calling new on the class.

`ngOnChanges`

> Invoked every time there is a change in one of th input properties of the component.

Reviewer: ```suggestion
Invoked every time there is a change in one of the input properties of the component.
```

`ngAfterContentChecked`
Invoked each time the content of the given component has been checked by the change detection mechanism of Angular.

`ngAfterViewInit`
Invoked when the component’s view has been fully initialized.To get more info about content query see official documentation [ViewChildren](https://angular.io/api/core/ViewChildren), [ViewChild](https://angular.io/api/core/ViewChild)

`ngAfterViewChecked`
Invoked each time the view of the given component has been checked by the change detection mechanism of Angular.

## Component Metadata

The `@Component` decorator identifies the class immediately below it as a component class, and specifies its metadata. The metadata for a component tells Angular where to get the major building blocks that it needs to create and present the component and its view. In particular, it associates a template with the component, either directly with inline code, or by reference. Together, the component and its template describe a view.

In addition to containing or pointing to the template, the @Component metadata configures, for example, how the component can be referenced in HTML and what services it requires.

#### Main properties

> `Selector`

Reviewer: All metadata props should be lowercased
```suggestion
`selector`
```
